src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -926,6 +948,28 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						reflectionContext.RecordUnrecognizedPattern ($"Activator call '{reflectionContext.MethodCalled.FullName}' inside '{reflectionContext.MethodCalling.FullName}' is not yet supported");
 						break;
 
+					//
+					// System.Runtime.CompilerServices.RuntimeHelpers
+					//
+					// RunClassConstructor (RuntimeTypeHandle type)
+					//
+					case IntrinsicId.RuntimeHelpers_RunClassConstructor: {
+							reflectionContext.AnalyzingPattern ();
+							foreach (var typeHandleValue in methodParams [0].UniqueValues ()) {
+								if (typeHandleValue is RuntimeTypeHandleValue runtimeTypeHandleValue) {
+									_markStep.MarkStaticConstructor (runtimeTypeHandleValue.TypeRepresented, new DependencyInfo (DependencyKind.AccessedViaReflection, reflectionContext.MethodCalling));
+									reflectionContext.RecordHandledPattern ();
+								} else if (typeHandleValue == NullValue.Instance)
+									reflectionContext.RecordHandledPattern ();
+								else {
+									reflectionContext.RecordUnrecognizedPattern ($"A {GetValueDescriptionForErrorMessage (typeHandleValue)} " +
```

Comment:
Nit/not really necessary: If we _really_ wanted - the `MemberKinds.Constructors` dataflow annotation keeps static constructors because they can be accessed with `typeof(Foo).GetConstructors(BindingFlags.NonPublic | BindingFlags.Static)`

Reply:
Good point. For now I would leave it as is. Supporting the annotation would mean to flow it through the get_TypeHandle call as well. Until we hit a use case I'm perfectly fine supporting just the simple constant cases.